docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,893 @@
+...
+```
+
+The inconsistency between the `IROHA2` and `IROHA` prefixes can be confusing. Of the two, the `IROHA` prefix is
+preferred, as users are discouraged from using Iroha 1 in any new projects.
+
+#### Inconsistent Naming of Configuration Parameters
+
+Current configuration demonstrates some inconsistencies in parameter naming, leading to potential user confusion. To
+address this, some refinements are proposed:
+
+- The `torii` section uses terms like `p2p_addr`, `api_url`, and `telemetry_url`. These are actually addresses in the
+  form of `localhost:8080` and not full URLs, which would include components like protocol and path. To standardize,
+  these should all be changed to use `addr` (e.g., `addr_p2p`, `addr_api`, `addr_telemetry`).
+- Consider moving addresses to their respective domain scopes. For example, instead of `torii.telemetry_url`, we could
+  have `telemetry.addr`.
+- In the `logger` section, the term `max_log_level` could be simplified to `log_level` or just `level` to make it more
+  concise without losing meaning. Although users are more likely to set this parameter using an environment variable.
+- In the `wsv` section, the field `wasm_runtime_config` has a redundant `_config` suffix. Given these are all
+  configuration parameters, such suffixes can be eliminated for clarity.
+
+Implementing a consistent naming convention like this will simplify the user's configuration process and reduce
+ambiguity.
+
+#### Redundant SCREAMING_CASE in JSON Configuration
+
+The adoption of SCREAMING_CASE for parameters within the JSON configuration appears out of place. Such uppercase field
+names might be customary for environment variables, but in a JSON setting, they don't offer any distinctive benefit.
+Here's a snippet from the current configuration showcasing this:
+
+```json
+{
+  "PUBLIC_KEY": null,
+  "PRIVATE_KEY": null,
+  "DISABLE_PANIC_TERMINAL_COLORS": false
+}
+```
+
+---
+
+In summary, these issues underscore the need for clearer, more intuitive naming conventions in Iroha 2's configuration.
+
+### Rustisms in the Configuration Reference
+
+Configuration reference contains Rust-specific expressions that wouldn't make sense to the end-user who is configuring
+Iroha and might not be familiar with Rust. Here are some examples:
+
+- [Using `Option<..>`](https://github.com/hyperledger/iroha/blob/35ba182e1d1b6b594712cf63bf448a2edefcf2cd/docs/source/references/config.md#option)
+- [Using `std::path::PathBuf`](https://github.com/hyperledger/iroha/blob/35ba182e1d1b6b594712cf63bf448a2edefcf2cd/docs/source/references/config.md#loggerlog_file_path)
+
+The use of rustisms like that overcomplicates the configuration reference and affects the user experience. Further, it
+doesn't actually provide any extra information to a user familiar with Rust, as environment variables and JSON values
+cannot be `None` due to being strictly typed and null-able regardless of whether the type specifies `Option`.
```

Comment:
```suggestion
cannot be `None` due to being stringly typed and null-able regardless of whether the type specifies `Option`.
```
Because this line got me confused TBH, and I believe that's what @appetrosyan said himself.